src/pages/DatasetRegistration.jsx

```diff
@@ -0,0 +1,1461 @@
+          phenotype: phenotype ? phenotype.propertyValue : '',
+          nrParticipants: nrParticipants ? nrParticipants.propertyValue : '',
+          description: description ? description.propertyValue : '',
+          datasetRepoUrl: datasetRepoUrl ? datasetRepoUrl.propertyValue : '',
+          researcher: researcher ? researcher.propertyValue : '',
+          principalInvestigator: pi ? pi.propertyValue : '',
+          dac: dac,
+          isValidName: validName
+        },
+      };
+    }
+    );
+
+    if (!fp.isEmpty(dataset.dataUse)) {
+      prefillDataUseFields(dataset.dataUse);
+    }
+  },[validateDatasetName]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      await init();
+      const notificationData = await NotificationService.getBannerObjectById('eRACommonsOutage');
+      const currentUser = await Storage.getCurrentUser();
+      const allDatasets = await DataSet.getDatasets();
+      const allDatasetNames = allDatasets.map(d => d.name);
+      const dacs = await DAC.list();
+
+      setRegistrationState(prevState => ({
+        ...prevState,
+        notificationData,
+        datasetData: {
+          ...prevState.datasetData,
+          researcher: currentUser.displayName
+        },
+        allDatasets,
+        allDatasetNames,
+        dacList: dacs
+      }));
+
+      if (!fp.isEmpty(registrationState.updateDataset)) {
+        prefillDatasetFields(registrationState.updateDataset);
+        const ontologies = await getOntologies(registrationState.formData.diseases);
+        const formattedOntologies = formatOntologyItems(ontologies);
+
+        setRegistrationState(prevState => ({
+          ...prevState,
+          formData: {
+            ...prevState.formData,
+            ontologies: formattedOntologies,
+            diseases: !fp.isEmpty(ontologies)
+          }
+        }));
+      }
+    };
+
+    fetchData();
+  }, [init, registrationState.formData.diseases, prefillDatasetFields, registrationState.updateDataset]);
+
+  const getOntologies = async (urls = []) => {
+    if (fp.isEmpty(urls)) {
+      return [];
+    } else {
+      const doidArr = OntologyService.extractDOIDFromUrl(urls);
+      const urlParams = doidArr.join(',');
+      const ontologies = await OntologyService.searchOntology(urlParams);
+      return ontologies;
+    }
+  };
+
+
+  // eslint-disable-next-line no-unused-vars
+  const handleOpenModal = () => {
+    setRegistrationState(prevState => ({
+      ...prevState,
+      showModal: true
+    }));
+  };
+
+  // eslint-disable-next-line no-unused-vars
+  const handleCloseModal = () => {
+    setRegistrationState(prevState => ({
+      ...prevState,
+      showModal: false
+    }));
+  };
+
+  const handleChange = (e) => {
+    const field = e.target.name;
+    const value = e.target.value;
+    setRegistrationState(prevState => ({
+      ...prevState,
+      datasetData: {
+        ...prevState.datasetData,
+        [field]: value
+      },
```

Comment:
```suggestion
      datasetData[field]: value
```

Would this work instead? The current syntax looks a bit confusing.

Reply:
Agreed. I made an update to this.